Add new comment hooks for comment author and meta data customization

DangitRick opened the pull request:
## Description
Creates two new hooks (`comment_author_data` and `comment_metadata`) within the `Comment_Walker` class, moves the default comment author/avatar and metadata output to two new comment template functions, and adds those functions to the new hooks. 

## Motivation and context
Customization should be as accessible as possible, and these new hooks will allow users to customize comment output even more thoroughly, without the need to outright replace the `Comment_Walker` functions wholesale using a callback on `wp_list_comments()` in a child theme's comments file, which is an awful lot to expect users to do in order to customize what info is shown with a comment.
Addresses this ticket (made on 1.x): https://github.com/ClassicPress/ClassicPress/issues/1246

I apologize for deleting my original pull request and am resubmitting it with slightly cleaner code.

## How has this been tested?
A slight variation of this code has been in use on my ClassicPress 1.x site for a while now, while this exact code is functioning without error or warning (with debug enabled) on a live 2.x test site.

## Types of changes
 - New Features (two new hooks)

DangitRick committed:
Create branch.

DangitRick committed:
Add comment author/meta customization hooks

Modifies the default comment walker by adding two new hooks — comment_author_data and comment_metadata — and moving the default output to comment template functions.

Theme authors and site owners can customize the display of comment authors, avatars, and date info by using a custom comment walker function (as before) or by now using these new hooks, which may be a lot more accessible to some.

DangitRick committed:
Revert "Create branch."

This reverts commit f652e156230ff15d1590e374e10799d4984d5aa0.

xxsimoxx reviewed (changes requested):
Please add DocBlock before `do action` to explain the two new hooks and to the two changed function to explain the change.

DangitRick committed:
Improved commenting

Adds comments explaining the new actions and where the default comment author/metadata went to.

xxsimoxx commented:
DocBlock comment should be in a specific format. I'm not so good with them so maybe @mattyrob can help.

mattyrob committed:
Improve docblocks

mattyrob commented:
I've pushed some docblock changes and hopefully fixed the failing test also.

viktorix commented on src/wp-includes/comment-template.php:
```diff
@@ -1174,6 +1174,78 @@ function comment_type( $commenttxt = false, $trackbacktxt = false, $pingbacktxt
+		);
+	}
+
+	current_theme_supports( 'html5' ) ? edit_comment_link( __( 'Edit' ), ' <span class="edit-link">', '</span>' ) : edit_comment_link( __( '(Edit)' ), ' &nbsp;&nbsp;', '' );
```
@mattyrob shouldn't we avoid ternary operators in the core?

DangitRick replied:
If so, I can push an update that swaps it for a traditional if/else at any time.

xxsimoxx replied:
Maybe it can added in the previous lines if/then/else block.

DangitRick replied:
Just pushed an update to the format that simplifies the if/else logic and removes the ternary logic altogether.

DangitRick committed:
Remove ternary format

Minor change to remove ternary operation and simplify if/else logic.

mattyrob commented:
Two approvals, merging for wider testing.
